### ISSUE TITLE
Enable people to sign up with email on pledge pages

### ISSUE THREAD

**Issue by aaronlifshin** [MayOneUS/teams#28]
Pledge pages should become recruitment pages for our core supporters/volunteers to be able to spread the word and get their friends to sign up and become supporters of Mayday as well.

We should be able to create leaderboards and contests, and keep track of who signed how many people up.

A

**Comment by aaronlifshin** [comment]
Email and optional additional information (e.g.: zipcode) should be collected and sent into the mailchimp endpoint.

Potentially we don't want to store them just in Mailchimp.

**Comment by davidxia** [comment]
@aaronlifshin let me know if I got the requirements for MVP correct:
- don't remove any existing stuff
- add a form to collect email and zipcode (anything else)
- send form data to mailchimp
- store a counter in Google App Engine (GAE) indicating how many signups a particular page got

**Comment by aaronlifshin** [comment]
Yep.  Pretty much spot on Spot on.

> store a counter in Google App Engine (GAE) indicating how many signups a
> particular page got
> More specifically, the counter can go in Teams repo on the team model.

The counter should also be displayed on the page.  Another (distinct) box
similar to where we have amount + number of pledges, that contains "Number
Recruited".

A

**Comment by davidxia** [comment]
@aaronlifshin Stubbed out the front end in the commit above. I assumed the email signups are the primary call to action and the donations are secondary. If not we can rearrange.

I wasn't able to test the form yet because of dependencies on the pledge service because show_team2.html [includes JS](https://github.com/davidxia/teams/blob/547532f7eceff4299066d4b8309d61da1be15159/templates/show_team2.html#L86) from the pledgeservice which is needed to actually [submit the pledge form](https://github.com/MayOneUS/pledgeservice/blob/1b2d0e38f1ca63546d81c739a993b276843374a9/assets/pledge.js#L298).

Cross repo deps like this seems like something to be avoided? If so, I'll just submit the form an endpoint handled by the teams repo?

**Comment by aaronlifshin** [comment]
I agree this is not the greatest dependency.  It's this SOA that we have
created, where pledgeservice is a service that is responsible for handling
payments.

I'm not sure of the right way to handle it.  We definitely need the
donation to be submitted to pledgeservice.  Are you suggesting that we
create an endpoint in teams that pushes to pledgeservice form the backend?

That js file has javascript interaction for the donate form.  This same
form exists in homepage_redesign repo in the sidebar as well as here in
teams, and they share this js, which we placed in pledgeservice.

I wonder if the best thing to do here would be to move the form to
pledgeservice as well and include in an iframe?  Ideally, all the donate
functionality should be in pledgeservice.

@jtolds had thoughts on how best to do this.

A

**Comment by davidxia** [comment]
@aaronlifshin I see the rationale.

Is there a way to get all the apps running at once on my local machine so that my dev environment is at least fully functional and I can test everything? It'd be nice to get a fast develop-test-debug cycle going on people's local machines to make it easy to contribute back.

**Comment by aaronlifshin** [comment]
This is tricky.  I have never gotten auth running locally properly, but
with the fake auth that teams uses, it's not too hard, I think to run teams
and pledgeservice together.

You just edit the config_NOCOMMIT.py to use this:

#PLEDGE_SERVICE_URL = "http://0.0.0.0:8080"

And then you install pledgeservice, and run that with grunt local.  That
should work, and you can make donations via test stripe.  Google test
stripe cards for the different tests.

A

**Comment by davidxia** [comment]
@aaronlifshin Thanks. I set `PLEDGE_SERVICE_URL = "http://0.0.0.0:8080"` in `teams: config_NOCOMMIT.py`. Then I started pledgeservice and then teams. But I see that when I request http://0.0.0.0:8080/r/total?team=foobar, this request is handled by teams and not pledgeservice.

I stopped both and tried starting teams first and then pledeservice. Same result.

When I stop teams, the request is handled by pledgeservice. When I start teams back up, it takes precedence again.

I guess the 2 apps cannot run side by side as dev_appserver.py seems to intercept all requests before the node one? I guess I can modify them to run on different ports.
